### DISCUSSION TITLE
Flicker on restart/redraw

### DISCUSSION THREAD

**Discussion by Davincible** [polybar/polybar#2758] [Q&A]
With either `polybar-msg cmd restart` or `-r` polybar can restart, but it looks as if it kills the bar, then re-reders it, causing a flickering effect. It even goes so far as to not take up any window space, causing my windows under it to resize for a split second until polybar redraws again. 

Is it possible for polybar to redraw 1. without visual flicker 2. without giving up window space?

**Comment by patrick96** [comment]
That's how restarting works in polybar. The entire program terminates and starts fresh, as if you killed polybarr and started it again.
